patch.sh

```diff
@@ -183,8 +183,15 @@ patch_common () {
         exit 1
     fi
 
-    if ! driver_version=$("$NVIDIA_SMI" --query-gpu=driver_version --format=csv,noheader,nounits | head -n 1) ; then
-        echo 'Something went wrong. Check nvidia driver'
+    cmd="$NVIDIA_SMI --query-gpu=driver_version --format=csv,noheader,nounits"
+    driver_versions_list=$($cmd)
+    ret_code=$?
+    driver_version=$(echo "$driver_versions_list" | head -n 1)
+    if [ $ret_code -ne 0 ] && [ -nz driver_version ] ; then
```

Comment:
shellcheck complains to this line:

```
[omnipotent@holy-see nvidia-patch]$ shellcheck patch.sh 

In patch.sh line 190:
    if [ $ret_code -ne 0 ] && [ -nz driver_version ] ; then
                                ^-^ SC2058: Unknown unary operator.

For more information:
  https://www.shellcheck.net/wiki/SC2058 -- Unknown unary operator.
```

Obviously, there is a syntax error. Also please note we use everywhere bash built-in `[[`. It will be nice if all code will look consistent.

I don't think we need second condition at all. If `nvidia-smi` exited with non-zero error code we can't expect it to return correct data.

Reply:
done